pkg/network/common/egressip.go

```diff
@@ -249,6 +252,17 @@ func (eit *EgressIPTracker) handleAddOrUpdateHostSubnet(obj, _ interface{}, even
 			return
 		}
 	}
+	if eit.recorder != nil && len(hs.EgressIPs) > 0 && eit.CloudEgressIP {
+		capacity, err := eit.getCapacity(hs)
+		if err != nil {
+			klog.Errorf("Ignoring invalid HostSubnet %s: %v", HostSubnetToString(hs), err)
+		}
+		if len(hs.EgressIPs) >= capacity {
+			hsRef := &corev1.ObjectReference{Kind: "HostSubnet", Name: hs.Name, UID: hs.ObjectMeta.UID}
+			eit.recorder.Eventf(hsRef, corev1.EventTypeNormal, "HostSubnet",
+				"The Hostsubnet reached its maximum node capacity %d", capacity)
```

Comment:
Maybe use the Node as the object reference rather than the HostSubnet? The HostSubnet is really just a weird implementation detail...